packages/ra-ui-materialui/src/auth/LoginForm.tsx

```diff
@@ -77,30 +105,35 @@ const LoginForm = ({ classes, isLoading, handleSubmit, translate }) => (
         </CardActions>
     </form>
 );
-LoginForm.propTypes = {
-    ...propTypes,
-    classes: PropTypes.object,
-    redirectTo: PropTypes.string,
-};
 
-const mapStateToProps = state => ({ isLoading: state.admin.loading > 0 });
+const mapStateToProps = (state: ReduxState) => ({
+    isLoading: state.admin.loading > 0,
+});
 
-const enhance = compose(
+const enhance = compose<Props & EnhancedProps, Props>(
     withStyles(styles),
-    translate,
+    withTranslate,
     connect(mapStateToProps),
     reduxForm({
         form: 'signIn',
-        validate: (values, props) => {
-            const errors = {};
+        validate: (values: FormData, props: TranslationContextProps) => {
+            const errors = { username: '', password: '' };
             const { translate } = props;
-            if (!values.username)
+            if (!values.username) {
                 errors.username = translate('ra.validation.required');
-            if (!values.password)
+            }
+            if (!values.password) {
                 errors.password = translate('ra.validation.required');
+            }
             return errors;
         },
     })
 );
 
-export default enhance(LoginForm);
+const LoginForm = enhance(LoginFormView);
+
+LoginForm.propTypes = {
```

Comment:
why is it still here?

Reply:
As explained in a previous comment, TypeScript does not generate propTypes and not everyone use TypeScript. PropTypes are still valuable